### PR TITLE
[Merged by Bors] - feat(Analysis/Analytic): define meromorphic functions

### PR DESCRIPTION
**Pull request by loefflerd**
This patch defines `MeromorphicAt`, for functions on nontrivially normed fields, and prove it is preserved by sums and products (routine) and inverses (much less trivial). Along the way, we define "order of vanishing" for a function analytic at a point (as an `ENat`).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Analytic/Meromorphic.lean [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  refine ⟨m + n, ?_⟩
+  convert hf.smul hg using 2 with z
+  rw [smul_eq_mul, ← mul_smul, mul_assoc, mul_comm (f z), ← mul_assoc, pow_add,
+    ← smul_eq_mul (a' := f z), smul_assoc, Pi.smul_apply']
+
+lemma mul {f g : 𝕜 → 𝕜} {x : 𝕜} (hf : MeromorphicAt f x) (hg : MeromorphicAt g x) :
+    MeromorphicAt (f * g) x :=
+  hf.smul hg
+
+lemma neg {f : 𝕜 → E} {x : 𝕜} (hf : MeromorphicAt f x) : MeromorphicAt (-f) x := by
```
Comment (alexjbest) [inline Mathlib/Analysis/Analytic/Meromorphic.lean:59]: Can we add the `iff` version as a simp lemma also?

**Comment by alexjbest** [comment]
This looks great to me. I'm sure there lots more handy API that can be added in the future, but this seems a reasonable place to start.
I'll leave it for a maintainer / reviewer that actually uses the analysis parts of the library to look also, but as far as I can tell this is ready

**Review by jcommelin — APPROVED** [review]
Looks good to me. Please add the `iff` version as Alex suggest, and merge afterwards.

bors d+

**Pull request closed without merging**